### ISSUE TITLE
Create a separate staging environment

### ISSUE THREAD

**Issue by jekkos** [opensourcepos/opensourcepos#2891]
As discussed, we'd like to have two environments on opensourcepos.org

* dev.opensourcepos.org
* demo.openourcepos.org

both can be hosted from docker-compose. The first will be the last pushed commit on any branch in the repo. The second will be the last release version (by parsing the package.json file in the repo itself).

This way we have a stable version for demo and an unstable (with possible issues) for testing and validation.

**Comment by mckaygerhard** [comment]
i think we nee reorganization so many many months ago.. as i started with the wiki .. but well .. i not work anymore in the wiki.. 

currently the installation proces are too complicated focused on security! .. security must be only after a pre.release are ready to freeze development and only testing commits are permited.. 

master brand in opensoucepos are "all kind of" commits. if a path or security fix must be done.. lasted time a rework of the git history was done .. incredible! pufff

if you wants to make that @jekkos  you must also have a stable brand and a development brand.. and current organization are far far away of that ! puff

**Comment by jekkos** [comment]
I have added another container that now serves the latest master

* demo.opensourcepos.org will always have the latest release version deployed. A deploy script checks github, fetches the version string and pulls the container from docker hub
* dev.opensourcepos.org will have the latest commit deployed (as we used to have on demo)

**Comment by jekkos** [comment]
The setup does use a separate mysql container for every install, this was a bit easier to configure. Personally I would have wanted to have two  schemas on one server but this scenario was not well supported in the mysql docker setup, so I took the easy route this time.

**Comment by chunter2** [comment]
@jekkos I'm noticing that dev.opensourcepos.org isn't currently at the latest commit in master.  Is there a specific time it gets updated or is this a manual process?  

Thanks

**Comment by jekkos** [comment]
Ok last push to docker hub was only 7 days ago, so that might explain..It could be the docker hub rate limitation blocking the update here.

**Comment by jekkos** [comment]
I have restarted the build so let's see if it refreshes the deploy. It's best to always check the version hash versus the repository when testing features..

**Comment by chunter2** [comment]
Yes, it's the version hash that helped me notice it wasn't up to date.  

Thanks

**Comment by jekkos** [comment]
It seems there is another issue (after the docker changes that were done). The container is not pushed to docker hub as it's not building it with docker-compose anymore. So I changed this setup now, let's see if it works.

**Comment by jekkos** [comment]
@chunter2 it's fine again now after my latest fix. Dev server is aligned to master again.
